examples/examples/runtime_update_async.rs

```diff
@@ -0,0 +1,106 @@
+/*
+	Copyright 2023 Supercomputing Systems AG
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+		http://www.apache.org/licenses/LICENSE-2.0
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+//! Example that shows how to detect a runtime update and afterwards update the metadata.
+use sp_keyring::AccountKeyring;
+use sp_weights::Weight;
+use substrate_api_client::{
+	ac_compose_macros::{compose_call, compose_extrinsic},
+	ac_primitives::{
+		AssetRuntimeConfig, Config, ExtrinsicSigner as GenericExtrinsicSigner, UncheckedExtrinsicV4,
+	},
+	api_client::UpdateRuntime,
+	rpc::JsonrpseeClient,
+	rpc_api::RuntimeUpdateDetector,
+	Api, SubmitAndWatch, SubscribeEvents, XtStatus,
+};
+use tokio::select;
+use tokio_util::sync::CancellationToken;
+
+type ExtrinsicSigner = GenericExtrinsicSigner<AssetRuntimeConfig>;
+type Hash = <AssetRuntimeConfig as Config>::Hash;
+
+#[cfg(feature = "sync-examples")]
+#[tokio::main]
+async fn main() {
+	println!("This example is for async use-cases. Please see runtime_update_sync.rs for the sync implementation.")
+}
+
+#[cfg(not(feature = "sync-examples"))]
+pub async fn send_code_update_extrinsic(
+	api: &substrate_api_client::Api<AssetRuntimeConfig, JsonrpseeClient>,
+) {
+	let new_wasm: &[u8] = include_bytes!("kitchensink_runtime.compact.compressed.wasm");
+
+	// this call can only be called by sudo
+	let call = compose_call!(api.metadata(), "System", "set_code", new_wasm.to_vec());
+	let weight: Weight = 0.into();
+	let xt: UncheckedExtrinsicV4<_, _, _, _> =
```

Comment:
```suggestion
	let xt =
```
Does it work without the type? I think it makes it look a little more complicated than it is.